app/src/main/java/com/kylecombes/micrepeater/ScreenSlidePagerAdapter.java

```diff
@@ -0,0 +1,24 @@
+package com.kylecombes.micrepeater;
+
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import androidx.viewpager2.adapter.FragmentStateAdapter;
+
+import com.kylecombes.micrepeater.ScreenSlidePageFragment;
+
+public class ScreenSlidePagerAdapter extends FragmentStateAdapter {
+    private static final int NUM_PAGES = 6;
+    public ScreenSlidePagerAdapter(FragmentActivity fa) {
+        super(fa);
+    }
+
+    @Override
+    public Fragment createFragment(int position) {
+        return new ScreenSlidePageFragment().newInstance(position);
```

Comment:
A better way to do this is `return ScreenSlidePageFragment.newInstance(position);`

`newInstance(position)` is already a static method, so you don't have to initialize an empty object (of type `ScreenSlidePageFragment`) to use the static method to create another object (of the same type `ScreenSlidePageFragment`).